sc_az_reindex_dag.py

```diff
@@ -51,26 +67,84 @@
 Tasks with custom logic are relegated to individual Python files.
 """
 
-GET_NUM_SOLR_DOCS_PRE = task_solrgetnumdocs(AZ_DAG, CONFIGSET, 'get_num_solr_docs_pre', conn_id=SOLR_CONN.conn_id)
+# Single task to provide message about which controller triggered the DAG
+REMOTE_TRIGGER_MESSAGE = BashOperator(
+    task_id="remote_trigger_message",
+    bash_command='echo "Remote trigger: \'{{ dag_run.conf["message"] }}\'"',
+    dag=DAG,
+)
 
-CREATE_COLLECTION = create_sc_collection(AZ_DAG, SOLR_CONN.conn_id, COLLECTION, REPLICATION_FACTOR, CONFIGSET)
+REQUIRE_DAG_RUN = PythonOperator(
+    task_id="require_dag_run",
+    python_callable=require_dag_run,
+    provide_context=True,
+    dag=DAG,
+)
+
+SET_COLLECTION_NAME = PythonOperator(
+    task_id="set_collection_name",
+    python_callable=datetime.now().strftime,
+    op_args=["%Y-%m-%d_%H-%M-%S"],
+    dag=DAG
+)
+
+GET_NUM_SOLR_DOCS_PRE = task_solrgetnumdocs(
+    DAG,
+    CONFIGSET,
+    "get_num_solr_docs_pre",
+    conn_id=SOLR_CONN.conn_id
+)
 
-INGEST_DATABASES = ingest_databases(dag=AZ_DAG, conn=SOLR_CONN, solr_url=SOLR_URL)
+CREATE_COLLECTION = tasks.create_sc_collection(
+    DAG,
+    SOLR_CONN.conn_id,
+    CONFIGSET + "-{{ ti.xcom_pull(task_ids='set_collection_name') }}",
+    REPLICATION_FACTOR,
+    CONFIGSET
+)
 
-GET_NUM_SOLR_DOCS_POST = task_solrgetnumdocs(AZ_DAG, COLLECTION, 'get_num_solr_docs_post', conn_id=SOLR_CONN.conn_id)
+INDEX_DATABASES = BashOperator(
+    task_id="index_databases",
+    bash_command=AIRFLOW_HOME + "/dags/cob_datapipeline/scripts/ingest_databases.sh ",
+    env={
+        "AZ_BRANCH": AZ_BRANCH,
+        "AZ_CLIENT_ID": AZ_CLIENT_ID,
+        "AZ_CLIENT_SECRET": AZ_CLIENT_SECRET,
+        "HOME": AIRFLOW_USER_HOME,
+        "SOLR_AUTH_PASSWORD": SOLR_CONN.password,
+        "SOLR_AUTH_USER": SOLR_CONN.login,
+        "SOLR_AZ_URL": tasks.get_solr_url(SOLR_CONN, CONFIGSET + "-{{ ti.xcom_pull(task_ids='set_collection_name') }}"),
+    },
```

Comment:
I believe this should be

```ruby
    env={**os.environ, **{
        "AZ_BRANCH": AZ_BRANCH,
        "AZ_CLIENT_ID": AZ_CLIENT_ID,
        "AZ_CLIENT_SECRET": AZ_CLIENT_SECRET,
        "HOME": AIRFLOW_USER_HOME,
        "SOLR_AUTH_PASSWORD": SOLR_CONN.password or "",
        "SOLR_AUTH_USER": SOLR_CONN.login or "",
        "SOLR_AZ_URL": tasks.get_solr_url(SOLR_CONN, CONFIGSET + "-{{ ti.xcom_pull(task_ids='set_collection_name') }}"),
    }},
```

Reply:
See above re: why we're polluting the BashOperator envvars with this Python native override.

Reply:
Because I'm still holding hope that we wont have to run `source ~/.bashrc` to get some of our scripts to work correctly.

Reply:
Also, see my note about environment that airflow sets... by not creating a new variable using os.environ we are actually removing variables which would be available in the overridden version